corehq/apps/hqwebapp/static/hqwebapp/js/tab_url_hashes.js

```diff
@@ -0,0 +1,36 @@
+/*
+ * Adds URL hash behavior to bootstrap tabs. This enables bookmarking/refreshing and browser back/forward.
+ * Lightly modified from https://stackoverflow.com/questions/18999501/bootstrap-3-keep-selected-tab-on-page-refresh
+ */
+hqDefine("hqwebapp/js/tab_url_hashes", [
+    "jquery",
+], function(
+    $
+) {
+    $(function(){
+        var tabSelector = "a[data-toggle='tab']:not(.no-hash)";
+        if (window.location.hash) {
+            $("a[href='" + window.location.hash + "']").tab('show');
+        } else {
+            $(tabSelector).first().tab('show');
+        }
+
+        $('body').on('click', tabSelector, function (e) {
+            e.preventDefault();
+            var tabName = this.getAttribute('href');
+            if (window.history.pushState) {
```

Comment:
Is this not a use of history? Re: commit 41e7ba0c7dbd86425dafe09394e1fc06a0c8925f (Remove history-1.7.1, since we aren't using History anywhere).

Reply:
No - usage of `window.history` is okay, it's the global var `History` created by [history.js](https://github.com/browserstate/history.js/#working-with-historyjs-directly) that we haven't been using.